### PR TITLE
Thoughts and questions

### PR DESCRIPTION
**Pull request by nmatthews-witekio**
A series of unrelated questions that we might wish to consider -- some less code-related than others.

### CODE REVIEW

File: include/iotconnect_lib.h [hunk 1]
```diff
@@ -11,6 +11,8 @@
 
 // SSL style configuration. You can use a different include
 #ifndef ITOTCONNECT_LIB_CONFIG_HEADER
+// FIXME this looks like a typo ITOTCONNECT_LIB_CONFIG_HEADER should be IOTCONNECT_LIB_CONFIG_HEADER
+// however, it's used in multiple repos, so needs thought before modifying?
```
Comment (nik-markovic) [inline include/iotconnect_lib.h:15]: I don't think that fixing this typo would affect any other project even if they tracked the latest on the branch. Most projects track a specific commit/label of this repo.

File: src/iotconnect_lib.c [hunk 2]
```diff
@@ -27,8 +27,12 @@ bool iotcl_init(IotclConfig *c) {
         IOTCL_LOG ("IotConnectLib_Configure: combined name (cpid + uuid) exceeded maximum value" IOTCL_NL);
         return false;
     }
+    // FIXME this is a shallow copy -- have potential dangling pointers from c here?
+    // Only works if c is static, or allocated -- not if its from some local variable?
     memcpy(&config, c, sizeof(config));
 
+    // FIXME this check looks superfluous? already checked c before it was copied -- maybe it's needed for a deep copy
```
Comment (nik-markovic) [inline src/iotconnect_lib.c:34]: You are correct. This check is redundant.

**Comment by nmatthews-witekio** [comment]
Reworked this.
Removed questions and made best effort to address things directly.

**Comment by nmatthews-witekio** [comment]
Weird -- thought I closed this?

**Pull request closed without merging**